chatblade/chat.py

```diff
@@ -107,7 +107,10 @@ def map_single(result):
 
 def build_client(config):
     if "OPENAI_API_AZURE_ENGINE" in os.environ:
-        return openai.AzureOpenAI(api_key=config["openai_api_key"])
+        return openai.AzureOpenAI(
+            api_key=config["openai_api_key"],
+            azure_deployment=os.environ.get("OPENAI_API_AZURE_ENGINE"),
```

Comment:
This is unnecessary. If you set `OPENAI_API_MODEL` to the value of `OPENAI_API_AZURE_ENGINE`, you get the same results as the code change here.

Reply:
This change is introduced because the official `openai-python` library [requires initialization with deployment name](https://github.com/openai/openai-python/blob/7f9e85017a0959e3ba07834880d92c748f8f67ab/src/openai/lib/azure.py#L193) for an azure client and such name [doesn't seem to get picked up](https://github.com/npiv/chatblade/blob/f67a3026cb1fa48bdc5e57aaa8c9809690529e15/chatblade/chat.py#L110) by `chatblade` when initializing it. Although error handling might become tricky.

Reply:
I wonder why it's working for me without your change…hmm

Reply:
I figured out why. The model I thought I was selecting isn't being used. I've been using the default deployment without realizing it. I need your patch!

Reply:
I didn't know that was possible, but glad you have figured it out.